rpc/src/rpc_service.rs

```diff
@@ -195,6 +200,14 @@ impl RpcRequestMiddleware {
             match path {
                 DEFAULT_GENESIS_DOWNLOAD_PATH => {
                     inc_new_counter_info!("rpc-get_genesis", 1);
+                    if self.ledger_path.join(ZSTD_GENESIS_ARCHIVE).exists() {
+                        // Redirect to zstd download path
+                        return Self::redirect(ZSTD_GENESIS_DOWNLOAD_PATH).into();
```

Comment:
redirecting here would prevent a node which has updated to supporting zstd genesis archives from serving to one which has not